### ISSUE TITLE
v 4 doesn't work without web workers?

### ISSUE THREAD

**Issue by pylnata** [ujjwalguptaofficial/JsStore#215]
# Title 

v 4 works only with web workers?

# Description 

I've just updated to v 4 from v 3. I create connection without web worker and get error in console 
TypeError: Cannot read property 'run' of undefined
    at Connection.push../node_modules/jsstore/dist/jsstore.commonjs2.js.module.exports.ConnectionHelper.sendRequestToWorker_ (jsstore.commonjs2.js:459)
    at Connection.push../node_modules/jsstore/dist/jsstore.commonjs2.js.module.exports.ConnectionHelper.executeQry_ (jsstore.commonjs2.js:432)
    at Connection.push../node_modules/jsstore/dist/jsstore.commonjs2.js.module.exports.ConnectionHelper.prcoessExecutionOfQry_ (jsstore.commonjs2.js:418)
    at jsstore.commonjs2.js:405

**Comment by ujjwalguptaofficial** [comment]
@pylnata - jsstore provide a plugin for without web worker connection in module system. have a look here - https://jsstore.net/tutorial/connection/#webpack-without-web-worker

also about 2nd regarding `select` api with web worker - i don't think that should be the issue. it might be you are doing something different. But anyway - could you please create a reproducable POC, so that i can have a look and fix.

**Comment by ujjwalguptaofficial** [comment]
Here is an example code - https://github.com/ujjwalguptaofficial/jsstore-examples/blob/master/typescript_without_worker/src/storage_service/idb_helper.ts

**Comment by pylnata** [comment]
Thanks @ujjwalguptaofficial . I am trying to follow by example. But probably because of my webpack config I get error:

```
client:159 ./node_modules/jsstore/dist/worker_injector/index.js
Module not found: Error: Can't resolve './jsstore.worker.commonjs2.js' in '/Users/pylnata/workspace/neo-ui/node_modules/jsstore/dist/worker_injector'
```

When I change path in /node_modules/jsstore/dist/worker_injector/index.js
from  module.exports = require('./jsstore.worker.commonjs2.js'); to module.exports = require('../jsstore.worker.commonjs2.js');  this error disappears.
 
But I get new one:
```
TypeError: Cannot read property 'run' of undefined
    at Connection.push../node_modules/jsstore/dist/jsstore.commonjs2.js.module.exports.ConnectionHelper.sendRequestToWorker_ (jsstore.commonjs2.js:459)
    at Connection.push../node_modules/jsstore/dist/jsstore.commonjs2.js.module.exports.ConnectionHelper.executeQry_ (jsstore.commonjs2.js:432)
    at Connection.push../node_modules/jsstore/dist/jsstore.commonjs2.js.module.exports.ConnectionHelper.prcoessExecutionOfQry_ (jsstore.commonjs2.js:418)
    at jsstore.commonjs2.js:405


```

**Comment by pylnata** [comment]
Hi @ujjwalguptaofficial I also tried to run your example  https://github.com/ujjwalguptaofficial/jsstore-examples/blob/master/typescript_without_worker/

It gives error in console:

VM40:1 Uncaught SyntaxError: Unexpected identifier
    at Object.Function [as setup] (<anonymous>)
    at Connection../node_modules/jsstore/dist/jsstore.commonjs2.js.module.exports.Connection.addPlugin (jsstore.commonjs2.js:742)
    at Module../src/storage_service/idb_helper.ts (idb_helper.ts:7)
    at __webpack_require__ (bootstrap:19)
    at Module../src/storage_service/base_service.ts (base_service.ts:1)
    at __webpack_require__ (bootstrap:19)
    at Module../src/storage_service/student_service.ts (idb_helper.ts:52)
    at __webpack_require__ (bootstrap:19)
    at Module../src/business/student_logic.ts (log.js:47)
    at __webpack_require__ (bootstrap:19)

Could you please take a look?

**Comment by ujjwalguptaofficial** [comment]
there was some issue of file path issue(i guess i did this to fix some issue and it broke here), have fixed this in new version - 4.0.6 and also updated the example with new version.

please check from your side and let me know. In order to install latest version, use this command - 

```
npm i jsstore@latest
```

**Comment by ujjwalguptaofficial** [comment]
yes we use the same code for both web worker & without web worker. Only in case of web worker, the file is used to initialize the web worker & file should be browser supported.

```
var worker = new Web Worker(file_name)
```
If you are runing jsstore with web worker, you will see a web worker symbol in dev tools something like below 

![Screenshot from 2021-05-25 17-12-38](https://user-images.githubusercontent.com/10938615/119492261-79ae6a00-bd7c-11eb-93ac-f4fa30af93da.png)

**Comment by pylnata** [comment]
Thank you very much @ujjwalguptaofficial  Yes I managed to run v 4 with and without web-workers. 
Interesting that now I have this issue with different result of the same query in both modes. 
I am going to investigate it more deeply as I really want to stay with 4 version, this library has really cool api.

**Comment by pylnata** [comment]
FYI I figured out the root of my problem, and  it doesn't related to the library.
